# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/hyprgraphics/hotfix-jpeg.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- hyprgraphics-0.1.1/src/image/formats/Jpeg.cpp.vanilla	2024-12-18 21:01:07.519999982 +0100
+++ hyprgraphics-0.1.1/src/image/formats/Jpeg.cpp	2024-12-18 21:02:08.403333314 +0100
@@ -26,12 +26,12 @@
     decompressStruct.err = jpeg_std_error(&errorManager);
     jpeg_create_decompress(&decompressStruct);
     jpeg_mem_src(&decompressStruct, bytes.data(), bytes.size());
-    jpeg_read_header(&decompressStruct, true);
+    jpeg_read_header(&decompressStruct, (boolean)true);
 
 #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
-    decompressStruct.out_color_space = JCS_EXT_BGRA;
+    decompressStruct.out_color_space = JCS_RGB;
 #else
-    decompressStruct.out_color_space = JCS_EXT_ARGB;
+    decompressStruct.out_color_space = JCS_RGB;
 #endif
 
     // decompress
